packages/concerto-core/lib/decoratormanager.js

```diff
@@ -252,6 +252,186 @@ class DecoratorManager {
         newModelManager.fromAst(decoratedAst);
         return newModelManager;
     }
+    /**
+    * Adds a key-value pair to a dictionary (object) if the key exists,
+    * or creates a new key with the provided value.
+    *
+    * @param {Object} dictionary - The dictionary (object) to which to add the key-value pair.
+    * @param {string} key - The key to add or update.
+    * @param {any} value - The value to add or update.
+    * @param {string} declaration - The target decl.
+    * @param {string} property The target property.
+    */
+    static addToDict(dictionary, key, value, declaration,property) {
+        const val = {
+            declaration,
+            property,
+            dcs: JSON.stringify(value),
+        };
+        if (dictionary[key]) {
+            dictionary[key].push(val);
+        } else {
+            dictionary[key] = [val];
+        }
+    }
+
+    /**
+     * Extracts all the decorator commands from all the models in modelManager
+     * @param {ModelManager} modelManager the input model manager
+     * @param {object} [options] - decorator models options
+     * @returns {Object} a new model manager with the decorations removed and a list of extracted decorator jsons
+     */
+    static extractDecorators(modelManager,options) {
+        const ast = modelManager.getAst(true);
+        const unDecoratedAst = JSON.parse(JSON.stringify(ast));
+        let decoractorDict={};
+        unDecoratedAst.models.forEach((model) => {
+            if (model.decorators && model.decorators.length>0){
+                this.processDecorators(model.namespace, model.decorators, decoractorDict, options.removeDecoratorsFromModel,'','');
+            }
+            model.declarations.forEach((decl) => {
+                this.extractDecoratorFromModel(model.namespace,decl,decoractorDict,options.removeDecoratorsFromModel);
+            });
+        });
+        const newModelManager = new ModelManager();
+        newModelManager.fromAst(unDecoratedAst);
+        let dcms=this.parseDecorators(decoractorDict);
+        let vocab = this.parseVocabs(decoractorDict,options.locale);
+        return {
+            modelManager:newModelManager,
+            decoratorCommandSet:dcms,
+            vocabularies:vocab
+        };
+    }
+
+    /**
+     * Parses the dict data into an array of decorator jsons
+     * @param {Object} decoratorDict the input dict
+     * @returns {Array<Object>} the parsed decorator command set array
+     */
+    static parseDecorators(decoratorDict){
```

Comment:
Since this is excluding vocabulary decorators, maybe it's name can be less generic, like `parseNonVocabularyDecorators`?